src/main/java/com/guysfromusa/carsgame/services/MovementsHistoryService.java

```diff
@@ -12,28 +18,59 @@
 import java.util.List;
 import java.util.Optional;
 
+import static com.guysfromusa.carsgame.entities.MovementsHistoryEntityBuilder.aMovementsHistoryEntity;
 import static org.apache.commons.lang3.Validate.notNull;
 
 /**
  * Created by Dominik Zurek, 26.02.18
  */
 
 @Service
-@Transactional(readOnly = true)
+@Transactional
 public class MovementsHistoryService {
 
+    private final CarRepository carRepository;
+    private final GameRepository gameRepository;
     private final MovementsHistoryRepository repository;
     private final ConversionService conversionService;
 
     @Autowired
-    public MovementsHistoryService(MovementsHistoryRepository repository,  ConversionService conversionService) {
+    public MovementsHistoryService(CarRepository carRepository,
+                                   GameRepository gameRepository,
+                                   MovementsHistoryRepository repository,
+                                   ConversionService conversionService) {
+        this.carRepository = notNull(carRepository);
+        this.gameRepository = notNull(gameRepository);
         this.repository = notNull(repository);
         this.conversionService = notNull(conversionService);
     }
 
+    @Transactional(readOnly = true)
     public List<MovementHistory> findMovementsHistory(List<String> gameIds, List<String> carNames, Optional<Integer> limitOfRecentStep) {
         List<MovementsHistoryEntity> movements = repository.findMovements(gameIds, carNames, limitOfRecentStep);
         return StreamUtils.convert(movements,
                 movementsHistoryEntity -> conversionService.convert(movementsHistoryEntity, MovementHistory.class));
     }
+
+    public void saveMove(String gameName, MoveResult moveResult) {
```

Comment:
storing all changes in one transaction would simplify and speedup data storing part